DigitalLearningSolutions.Web/Scripts/support/faqs.ts

```diff
@@ -0,0 +1,12 @@
+import { SearchSortFilterAndPaginate } from '../searchSortFilterAndPaginate/searchSortFilterAndPaginate';
+
+(function dlsSupportFaqsHelper(): void {
+  const matches = window.location.pathname.match(/^\/(\w+)\/Support\/FAQs#?$/);
+
+  if (!matches || matches.length < 2) {

```

Comment:
Match here brings back an array of two results, the original string and the match.

Reply:
I can do but my preference would to not do that given what is happening here is just the standard behaviour of this built in JS method. 
What would the comment give beyond what can be understood from existing documentation?
Although note, where I said "original string" before that should actually read "the whole regex match".

I suppose what could be done if we want more clarity over what is happening and expected then I could re-write this to something more like the below given we are looking for an exact match:

```  
const pathMatchResults = window.location.pathname.match(/^\/(?<SubApplication>\w+)\/Support\/FAQs#?$/);
if (pathMatchResults === null) {
  return;
}
const subApplication = pathMatchResults.groups.SubApplication;
// eslint-disable-next-line no-new
new SearchSortFilterAndPaginate(`${subApplication }/Support/FAQs/AllItems`, true, true, false, undefined, ['title', 'content']);
```

Reply:
Yeah, nice, that refactor does make it clearer. One question and two thoughts which would make it even clearer:
* What happens if no SubApplication group is found?
* You could rename `dlsSupportFaqsHelper` to `InitiateFaqsPagination` or something to that effect
* You could extract the code that gets the dlsSubApplication into a function called `getDlsSubApplication()`

I agree that a comment isn't needed.

Reply:
Regarding no sub-application:
This JS should only be run on one of the two related pages - this code is either going to work because it is called in the right place, or not work if it is not called in the right place.
So any error should effectivaly highlight this fact.

For the other points, I've sorted those following your suggestion, although in theory the function does not really need a name at all but the names help.

Reply:
Thanks, yeah, just so that people reading the file understand what the js is doing. Looks good!